homeassistant/components/onkyo/media_player.py

```diff
@@ -120,6 +120,9 @@
 
 def _parse_onkyo_tuple(tup):
     """Parse a tuple returned from the eiscp library."""
+    if isinstance(tup, bool):
```

Comment:
can't this be paired with the next if ?

Reply:
it could, but since these are different cases I thought this was slightly more readable. No strong preference though.

Reply:
This point is now moot - the different cases require different behavior.

Reply:
This check should preferably be done before passing the argument to this function. It's generally better to check arguments before calling functions and have as strict parameter typing as possible.

Reply:
The objective of this method is to parse the returned information from the API. Unfortunately, the API can return different types. I can see a justification for splitting lines 129-132 into a function that will be called by the outer one, but I don't think it makes sense to make each caller responsible for checking the arguments.

Reply:
At least then update the parameter name and function docstring as currently they don't make sense.